Multisig/Data/Services/Networking Utils/HTTPClient.swift

```diff
@@ -104,11 +119,16 @@ public class HTTPClient {
         if let error = result.error {
             throw error
         }
-        guard let httpResponse = result.response as? HTTPURLResponse, (200...299).contains(httpResponse.statusCode),
-            let data = result.data else {
-                throw Error.networkRequestFailed(request, result.response, result.data)
+        guard let httpResponse = result.response as? HTTPURLResponse else {
+            throw Error.networkRequestFailed(request, result.response, result.data)
         }
-        return data
+        if (200...299).contains(httpResponse.statusCode) {
+            return result.data ?? Data()
+        } else if httpResponse.statusCode == 404 {
+            throw Error.entityNotFound(request, result.response, result.data)
+        } else if httpResponse.statusCode == 422 {
```

Comment:
great! can we have integration tests for each of these cases?

Reply:
added integration tests